gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -335,12 +364,17 @@ void connectHelixManager() {
    * the job with EXAMPLE_INSTANCE_TAG will remain in the ZK until an instance with EXAMPLE_INSTANCE_TAG was found.
    */
   private void addInstanceTags() {
-    if (this.helixManager.isConnected()) {
-      List<String> tags = ConfigUtils.getStringList(this.config, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY);
-      logger.info("Adding tags binding " + tags);
-      tags.forEach(tag -> helixManager.getClusterManagmentTool().addInstanceTag(this.clusterName, this.helixInstanceName, tag));
-      logger.info("Actual tags binding " + helixManager.getClusterManagmentTool()
-          .getInstanceConfig(this.clusterName, this.helixInstanceName).getTags());
+    List<String> tags = ConfigUtils.getStringList(this.config, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY);
+    HelixManager receiverManager = taskDriverHelixManager.isPresent()?taskDriverHelixManager.get()
```

Comment:
This seems to be a common pattern and is seen three times. Can you make a method for this like getReceiverManger()?